### ISSUE TITLE
String literals are not properly checked (Lua, C#, others?)

### ISSUE THREAD

**Issue by fghber** [Predelnik/DSpellCheck#166]
Hi. This is related to a question on [StackOverflow](https://stackoverflow.com/questions/54137038/extend-lua-language-settings-in-notepad-to-spell-check-also-stuff-enclosed-in/54141844#54141844) regarding DsSpellChecking's fails to check the spelling of multi-line strings in Lua. 

I have investigated a bit and came to the conclusion the underlaying problem is:   
String literals of certain programming languages like Lua (and C#) are not spell checked by the engine.
See: https://stackoverflow.com/a/54141844/8291949
The problem, I believe, was introduced in DsSpellCheck 1.4.x and related to the new feature to only check certain properties of a programming language, like strings, comments, function names.

**Comment by Predelnik** [comment]
Yes, unfortunately it's not easy to test every language thoroughly, I guess I could have noticed C# though I even programmed in it a bit.
Thanks for reporting, I will fix string literal checking for those two languages in the next version.

**Comment by fghber** [comment]
Awesome, thank you. If you can point out how this is fixed I could take a look into some more languages I know.

**Comment by Predelnik** [comment]
Well it's not that hard technically but code doing this stuff is a bit unseemly:
https://github.com/Predelnik/DSpellCheck/blob/master/src/SciUtils.cpp

Basically it's a giant switch for each language with nested switch for each style as their named in `SciLexer.h` file from Scintilla (which is engine Notepad++ utilizes).

The biggest problem is that naming of these constants is rather arbitrary and I can't even figure out which corresponds to C#  right now for example. So mostly it results in trial and error with some initial logic based on naming of constants sadly. All this stuff is not that well digestible for plugin usage obviously.

**Comment by fghber** [comment]
Cool. I got the idea. I can go ahead an create a PR to add `SCE_LUA_LITERALSTRING` as s::string`` (and others if I can find more).

**Comment by Predelnik** [comment]
@fghber I plan to release the next version soon-ish so I went ahead and applied fixes suggested by you. But certainly feel free to do pull request if you find any other problems in this or other part of the plugin. And of course big thanks for finding the required constants!

I also added styles in other languages which contained `VERBATIM` to be treated as string, but weren't able to thoroughly check if it changed much.
